### ISSUE TITLE
`tf.keras.losses.categorical_crossentropy` lack type check for `axis`

### ISSUE THREAD

**Issue by ArrowIntoTheSky** [keras-team/keras#16192]
**System information**.
- Have I written custom code (as opposed to using a stock example script provided in Keras): Yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Ubuntu 20.04
- TensorFlow installed from (source or binary): binary
- TensorFlow version (use command below): 2.8.0
- Python version: 3.8
- Bazel version (if compiling from source): n/a
- GPU model and memory: n/a

**Describe the problem**.

Describe the problem clearly here. Be sure to convey here why it's a bug in Keras or why the requested feature is needed.

**Standalone code to reproduce the issue**.
```
import tensorflow as tf
y_true = [[0, 1, 0], [0, 0, 1]]
y_pred = [[0.05, 0.95, 0], [0.1, 0.8, 0.1]]
from_logits = True
axis = True
loss1 = tf.keras.losses.categorical_crossentropy(y_true, y_pred)
loss2 = tf.keras.losses.categorical_crossentropy(y_true, y_pred, axis=axis) # Pass
loss3 = tf.keras.losses.categorical_crossentropy(y_true, y_pred, from_logits=from_logits, axis=axis) #AssertionError
```

**Describe the current behavior**.
See the code above. On the last line, `tf.keras.losses.categorical_crossentropy` will raise an `AssertionError` because `axis` cannot be a `bool` value. However, the computation of `loss2` would succeed. The only difference is that the third invocation of `tf.keras.losses.categorical_crossentropy` sets the `from_logits` to `True`. This does not make sense at all.

**Describe the expected behavior**.
Expect `tf.keras.losses.categorical_crossentropy(y_true, y_pred, axis=True)` to also raise a similar `AssertionError`.

**Comment by sachinprasadhs** [comment]
Internally these boolean values will be converted to binary, if axis it set to `True` it will be converted to 1 and same will be considered for operation, if axis is set to `False`, it would consider 0 as value. 
Anyways, I will send a fix to avoid confusion among the users. Thanks!

**Comment by ArrowIntoTheSky** [comment]
> 

@sachinprasadhs Thanks for the PR. The problem here is that when calling `tf.keras.losses.categorical_crossentropy(y_true, y_pred, from_logits=from_logits, axis=axis)`, it **will not convert** `True` to `1` but raise an **error**. In `tf.keras.losses.categorical_crossentropy(y_true, y_pred, axis=axis)`, it will automatically convert `True` to `1`. This inconsistency can be very confusing or even bring unexpected errors if users assume automatic conversion in all cases.
